### DISCUSSION TITLE
Works with

### DISCUSSION THREAD

**Discussion by hepfister** [robbinjanssen/home-assistant-omnik-inverter#98] [General]
I can confirm that this plugin works with the Omniksol-1.5k-TL (Java) and the Omniksol-3k-TL (Java). You can update the information page accordingly.

**Comment by klaasnicolaas** [comment]
Thanks for the info, we'll add them to the list.

**Comment by Videodok** [comment]
> I can confirm that this plugin works with the Omniksol-1.5k-TL (Java) and the Omniksol-3k-TL (Java). You can update the information page accordingly.

Apparently I'm doing something wrong, when you are having a working device with Java. as there is not much to fill in, the only thing I can think of, is that i'm not using the correct name, I assumed this is the account name configured in the device, bit this is not working for me.

**Reply by MarijnS95** [reply]
(Technical nit: Java is something completely different than JavaScript :grimacing:)

**Comment by PetersPlace** [comment]
I would like to thank you for your commitment. I can also report that on my Omnik Omniksol 4000TL2 the TCP connection also works.
Then you also get many more values ​​from your inverter such as voltage, current per string and many others.

**Comment by cyberjunky** [comment]
Was looking for this, thanks! (have a Omniksol-1.5k-TL)

**Comment by Jon78** [comment]
I can confirm that the Hosola Smart 2500 TL works, at least with TCP, possibly other ways as well. Look for Igen Wifi in your router logs and use the logger SN.

**Comment by mondo2020** [comment]
Used the integration with Bosswerk MI600 which is equal to Deye.
Works direct! Many thanks!

**Comment by gerritanne** [comment]
I can confirm this works with the Hosola 3000TL-S as well. Would be great if you guys could add it to the list (…I only saw the hosola 1500 inverter on the list, but I see in this thread that the 2500 TL is also working…)

Kind regards,
Gerrit

**Comment by DS723** [comment]
For what its worth...  my 10y old  Omnik 5000tl2 give some data returned in HA via HACS  on the TCP level.   Thanks for writing this peace of coding. Appreciated
